### DISCUSSION TITLE
Improving YimMenu

### DISCUSSION THREAD

**Discussion by Yimura** [YimMenu/YimMenu#1286] [Announcements]
To improve the stability of YimMenu we won't be accepting any PR's that add features to YimMenu for the foreseeable future.
(except for the PR's that are still open)

The goal is to restructure the UI in certain places that have become too clogged or merge certain submenu's that only contain one feature or nearly no features.

There's also the issues with certain missions being unplayable and just outright breaking.
So instead we'll be focussing our efforts to fixing those bugs.

## For the future

PR's need to be focussed on single features only, unless they're one big cohesive feature that's going to be added.
Having over a thousand lines of code to review makes it easy to miss certain small things.

Any PR's that add too many different features at once and do not follow the CONTRIBUTING guidelines will be outright denied.

Here's a good YT video to watch to give some insights on how the code review process goes:
https://youtu.be/1Ge__2Yx_XQ

**Comment by ghost** [comment]
Can the reactions submenu get reworked to something like this: 
![yimmenu_reactions_ui_change_proposal](https://user-images.githubusercontent.com/122435102/235728761-45978018-ef57-42ba-9d9a-e084273219f7.png)
?

**Reply by ghost** [reply]
that seems overkill and would take up alot of space

**Reply by Yimura** [reply]
I think it's fine really, the part in regards to protections, notifications and reactions is pretty cluttered.

**Reply by ghost** [reply]
> that seems overkill and would take up alot of space

yimmenu already takes up a lot of space. at least with this layout there isn't the requirement to first expand/collapse each and every option to even modify it. all of these options are already collapsed into their own submenu. they should not be collapsed further. and with the addition of check/uncheck all options that would make this submenu much less tedious to set to the end users preferences. with this layout you could even implement the option to enable/disable a whole row by clicking on the option name. 

with regard to screen real estate, i'm all for yimmenu taking up less space. that's a foundational problem for this menu and this proposal is only for one section of the menu. once that issue is decided and implemented the whole ui would probably require a redesign.

**Comment by iwanttodo1t** [comment]
yimmenu should first have a good "framework", and then let contributors develop features under the framework, so that yimmenu can better manage these functions, and can also make a scripting API so that those of us who don't know game development can also make interesting functions.

**Reply by iwanttodo1t** [reply]
The “framework” may not be accurate, I'm not a professional developer and don't quite understand what this form belongs to

**Reply by xiaoxiao921** [reply]
LUA scripting and a lua documented api is coming, but its gonna take time

**Comment by vincentix** [comment]
I think I speak for everyone following this project that we appreciate what you and the other contributors are doing here very much. :)
